issue #770: remove emoji keyboard from ios

IlarionHalushka opened the pull request:
@RocketChat/ReactNative
Closes #770 

Now on IOS there is no Emoji keyboard and "+"(plus) icon is moved to the left: 
https://prnt.sc/n8mk9g
https://prnt.sc/n8mkbp

On Android the layout is not changed:
https://prnt.sc/n8mucc 

Note: RCTCustomInputController.xcodeproj and KeyboardTrackingView.xcodeproj are not removed because removing any of them breaks the KeyboardAccessoryView from [react-native-keyboard-input](https://github.com/wix/react-native-keyboard-input#ios) package. Proof: https://prnt.sc/n8a4gx   https://prnt.sc/n8a3z7

diegolmello commented:
> Note: RCTCustomInputController.xcodeproj and KeyboardTrackingView.xcodeproj are not removed because removing any of them breaks the KeyboardAccessoryView from react-native-keyboard-input package. Proof: https://prnt.sc/n8a4gx https://prnt.sc/n8a3z7

The idea is to remove react-native-keyboard-input from iOS.
Android:
```jsx
<KeyboardAccessoryView>
  <Emoji />
  <TextInput />
  <Audio />
  <Plus />
</KeyboardAccessoryView>
```

iOS:
```jsx
<React.Fragment>
  <Plus />
  <TextInput />
  <Audio />
</React.Fragment>
```

IlarionHalushka commented:
I see, ok, thanks!

IlarionHalushka commented:
@diegolmello addressed your comments: removed native iOS libs and used KeyboardAvoidingView. Android behavior not changed.

iOS screenshots: 
![image](https://user-images.githubusercontent.com/25078524/56080717-4531fb00-5e0d-11e9-92b1-81b60a3bb5c3.png)
![image](https://user-images.githubusercontent.com/25078524/56080718-495e1880-5e0d-11e9-8c57-88b876798e6f.png)

Android screenshot:
![image](https://user-images.githubusercontent.com/25078524/56080721-524eea00-5e0d-11e9-88af-869f2eeb62d4.png)

Tested the main functionality and it seems to be working properly on both platforms: 
iOS:

- 	+ Right Buttons (send\audio)
		+ type text -> SEND button
		+ erase text -> AUDIO button
		+ send text
		+ audio record
		+ audio send
		+ audio cancel

- 	+ Text input
		+ type message
		+ mention someone (with #, with @)
		+ emoji
		+ message with paragraphs 
		+ reply message
		+ keyboard show/hide(on click out)

- 	+ Left buttons (plus)
		+ attach image
		+ cancel action sheet	

Android:
- 	+ Right Buttons (plus, send\audio)
		+ type text -> SEND button
		+ erase text -> PLUS button, AUDIO button
		+ send text
		+ audio record
		+ audio send
		+ audio cancel
		+ attach image
		+ cancel action sheet
- 	+ Text input
		+ type message
		+ mention someone (with #, with @)
		+ message with paragraphs 
		+ reply message
		+ keyboard show/hide(on click out)

- 	+ Left buttons (emoji)
		+ show emoji keyboard
		+ hide emoji keyboard
		+ send emoji
		+ send custom emoji
		+ change tabs in emoji picker

diegolmello commented:
@IlarionHalushka Really nice job! I'll take a look soon. Thanks.

diegolmello commented on app/containers/IOSKeyboardAvoidingView.js:
```diff
@@ -0,0 +1,46 @@
+		const { statusBarHeight } = this.state;
+
+		return (
+			<KeyboardAvoidingView
```
I didn't like the experience.
I know you removed because I told you so, but let's stick with https://github.com/wix/react-native-keyboard-tracking-view.

Just to summarize:
- iOS: will use https://github.com/wix/react-native-keyboard-tracking-view
- Android: will use https://github.com/wix/react-native-keyboard-input and https://github.com/wix/react-native-keyboard-tracking-view

diegolmello commented on app/containers/MessageBox/MessageBoxContent.android.js:
```diff
@@ -0,0 +1,183 @@
+import React from 'react';
```
You should merge MessageBoxContent into one file with both iOS and Android.
Too many duplicated code.

diegolmello commented on app/containers/MessageBox/index.js:
```diff
@@ -757,30 +660,24 @@ class MessageBox extends Component {
-						{this.rightButtons}
-					</View>
-				</View>
+				<MessageBoxContent
```
![image](https://user-images.githubusercontent.com/804994/56318180-0796e000-6135-11e9-89fe-89d8d1277ff9.png)

diegolmello commented:
I'll refactor Messagebox in a few days and I'll close this.
I'll try to cherry pick some of your commits.
